src/convenience/frameworks.ts

```diff
@@ -178,13 +179,34 @@ const http: FrameworkAdapter = (req, res) => {
 };
 
 /** AWS lambda serverless functions */
-const awsLambda: FrameworkAdapter = (event, _context, callback) => ({
-    update: JSON.parse(event.body),
-    header: event.headers[SECRET_HEADER],
-    end: () => callback(null, { statusCode: 200 }),
-    respond: (json) => callback(null, { statusCode: 200, body: json }),
-    unauthorized: () => callback(null, { statusCode: 401 }),
-});
+const awsLambda: FrameworkAdapter = (event, _context, callback) => {
+    let resolveResponse: (response: unknown) => void;
+    let handlerReturn: Promise<unknown> | undefined;
+
+    if (callback) {
```

Comment:
AWS lambdas [support both ways](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html), so that is not uncommon for middleware, plugins, and extensions aimed at it.

My intention was to keep the legacy behavior while adding support for the modern format.

That being said, separating it as a new adapter is also an alternative. Maybe call it `aws-lambda-async`?

Reply:
Yes. For every call signature there's exactly one adapter.

For comparison, check the two cloudflare adapters. One is for module workers, and one is for legacy workers.

Reply:
Does the current one have a problem with secret tokens? You can fix it if that's the case. If there's no problem, there's no need to touch the code.